zk/check.py

```diff
@@ -280,14 +282,13 @@ def parse_stat(self, buf):
         # body correctly. Particularly, the Connections val was added in
         # >= 3.4.4.
         start_line = buf.readline()
-        match = self.version_pattern.match(start_line)
+        match = self.version_pattern.search(start_line)
         if match is None:
             return (None, None, "inactive", None)
             raise Exception("Could not parse version from stat command output: %s" % start_line)
         else:
-            version_tuple = match.groups()
-        has_connections_val = version_tuple >= ('3', '4', '4')
-        version = "%s.%s.%s" % version_tuple
+            version = match.group()
+        has_connections_val = LooseVersion(version) > LooseVersion("3.4.4")
```

Comment:
Probably beyond the scope of this PR, but I hear we should probably be using `packaging.version.parse` instead as it's conformant to [PEP440](https://www.python.org/dev/peps/pep-0440/) whereas `distutils.version` conforms to the deprecated [PEP386](http://www.python.org/dev/peps/pep-0386/).